From 3458c5272045ec3d9a9ecc8764773f00d702bb1f Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Fri, 28 Jul 2023 16:10:01 +0000
Subject: [PATCH 220/222] v4l2_m2m_dec: Avoid structure init warnings when
 struct changes

(cherry picked from commit 8a836af420ed8c8dba90e2fd88691bcaa0668f8a)
---
 libavcodec/v4l2_m2m_dec.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/libavcodec/v4l2_m2m_dec.c b/libavcodec/v4l2_m2m_dec.c
index cec98cc16a..10165c1504 100644
--- a/libavcodec/v4l2_m2m_dec.c
+++ b/libavcodec/v4l2_m2m_dec.c
@@ -1012,11 +1012,13 @@ parse_extradata(AVCodecContext * const avctx, V4L2m2mContext * const s)
 #if CONFIG_H264_DECODER
         case AV_CODEC_ID_H264:
         {
-            H264ParamSets ps = {{NULL}};
+            H264ParamSets ps;
             int is_avc = 0;
             int nal_length_size = 0;
             int ret;
 
+            memset(&ps, 0, sizeof(ps));
+
             ret = ff_h264_decode_extradata(avctx->extradata, avctx->extradata_size,
                                            &ps, &is_avc, &nal_length_size,
                                            avctx->err_recognition, avctx);
@@ -1042,12 +1044,15 @@ parse_extradata(AVCodecContext * const avctx, V4L2m2mContext * const s)
 #if CONFIG_HEVC_DECODER
         case AV_CODEC_ID_HEVC:
         {
-            HEVCParamSets ps = {{NULL}};
-            HEVCSEI sei = {{{{0}}}};
+            HEVCParamSets ps;
+            HEVCSEI sei;
             int is_nalff = 0;
             int nal_length_size = 0;
             int ret;
 
+            memset(&ps, 0, sizeof(ps));
+            memset(&sei, 0, sizeof(sei));
+
             ret = ff_hevc_decode_extradata(avctx->extradata, avctx->extradata_size,
                                            &ps, &sei, &is_nalff, &nal_length_size,
                                            avctx->err_recognition, 0, avctx);
-- 
2.45.2

